src/lib/remark-plugin-adjust-link-urls.ts

```diff
@@ -8,7 +8,7 @@ const remarkPluginAdjustLinkUrls: Plugin = ({
 	urlAdjustFn: (url: string) => string
 }): Transformer => {
 	return function transformer(tree) {
-		visit(tree, 'link', (node: Link) => {
+		visit(tree, ['link', 'definition'], (node: Link) => {
```

Comment:
🙋🏻 Why is this update needed?

Reply:
Great question - it's needed to adjust URLs for all ways that links can be written in markdown. 

In detail: We currently only target `mdast` [`link` nodes only](https://github.com/syntax-tree/mdast#link), where the URL is written inline (like `[link text](https://www.example.com)`. We were not adjusting the URLs [`linkReference` nodes](https://github.com/syntax-tree/mdast#linkreference) (written like `[link text][reference-id]`), which reference [`definition` nodes](https://github.com/syntax-tree/mdast#definition). 

Related: we do something similar in the [`rewriteTutorialLinksPlugin`](https://github.com/hashicorp/dev-portal/blob/aa5aec86ace8004bf7b60d344308efd8af87b21e/src/lib/remark-plugins/rewrite-tutorial-links/index.ts#L75).